src/ethereum/EthHashInfo/index.tsx

```diff
@@ -70,42 +71,62 @@ const EthHashInfo = ({
   shortenHash,
   showAvatar,
   customAvatar,
+  customAvatarFallback,
   avatarSize = 'md',
   showCopyBtn,
   menuItems,
   explorerUrl,
-}: Props): React.ReactElement => (
-  <StyledContainer className={className}>
-    {showAvatar && (
-      <AvatarContainer>
-        {customAvatar ? (
-          <StyledImg src={customAvatar} size={avatarSize} />
-        ) : (
-          <Identicon address={hash} size={avatarSize} />
-        )}
-      </AvatarContainer>
-    )}
+}: Props): React.ReactElement => {
+  const [fallbackToIdenticon, setFallbackToIdenticon] = useState(false);
+
+  const setAppImageFallback = (
+    error: SyntheticEvent<HTMLImageElement, Event>
+  ): void => {
+    if (customAvatarFallback && !fallbackToIdenticon) {
+      error.currentTarget.onerror = null;
+      error.currentTarget.src = customAvatarFallback;
```

Comment:
Maybe we could avoid manipulating DOM directly?
```
setFallbackSrc(customAvatarFallback);

...

<StyledImg
  src={fallbackSrc || customAvatar}
  ...
/>
```

Reply:
you're right, I copied it directly from SRC.
what do you suggest? `ref` perhpahs?

Reply:
Just a state variable, see the code snippet in my previous comment.